entity_gym/entity_gym/environment.py

```diff
@@ -184,7 +203,67 @@ class ObsBatch:
     end_of_episode_info: Dict[int, EpisodeStats]
 
 
-def batch_obs(obs: List[Observation]) -> ObsBatch:
+def merge_obs(a: Optional[ObsBatch], b: ObsBatch) -> ObsBatch:
+    """
+    merges two obs_batches
+    """
+
+    if a == None:
+        return b
+
+    assert isinstance(a, ObsBatch)
+
+    assert (
+        a.entities.keys() == b.entities.keys()
+    ), "Malformed batches. batches must contain the same entity keys to be merged"
+    assert (
+        a.action_masks.keys() == b.action_masks.keys()
+    ), "Malformed batches. batches must contain the same action mask keys to be merged"
+
+    entities = {}
+    ids: Sequence[Sequence[EntityID]] = []
+    action_masks: Dict[
+        str, Union[CategoricalActionMaskBatch, SelectEntityActionMaskBatch]
+    ] = {}
+    reward = []
+    done = []
+    end_of_episode_info: Dict[int, EpisodeStats] = {}
+
+    # merge entities
+    for k in a.entities.keys():
+        entities[k] = a.entities[k]
+        entities[k].extend(b.entities[k])
+
+    # merge ids
+    ids.extend(a.ids)  # type: ignore
+    ids.extend(b.ids)  # type: ignore
+
+    # merge masks
+    for k in a.action_masks.keys():
+        action_masks[k] = a.action_masks[k]
+        action_masks[k].extend(b.action_masks[k])
```

Comment:
This mutates `a`. We should either create a copy of `action_masks[k]` here, or change everything to just mutate `a` and make that more obvious in the type signature (by returning `None`).